asserts/database.go

```diff
@@ -262,20 +265,35 @@ func (db *Database) IsTrustedAccount(accountID string) bool {
 
 // Check tests whether the assertion is properly signed and consistent with all the stored knowledge.
 func (db *Database) Check(assert Assertion) error {
-	// TODO: later may need to consider type of assert to find candidate keys
-	accKey, err := db.findAccountKey(assert.AuthorityID(), assert.SignKeyID())
-	if err == ErrNotFound {
-		return fmt.Errorf("no matching public key %q for signature by %q", assert.SignKeyID(), assert.AuthorityID())
-	}
-	if err != nil {
-		return fmt.Errorf("error finding matching public key for signature: %v", err)
-	}
-
-	now := time.Now()
-	for _, checker := range db.checkers {
-		err := checker(assert, accKey, db, now)
+	typ := assert.Type()
+	if typ.flags&noAuthority == 0 {
+		// TODO: later may need to consider type of assert to find candidate keys
+		accKey, err := db.findAccountKey(assert.AuthorityID(), assert.SignKeyID())
+		if err == ErrNotFound {
+			return fmt.Errorf("no matching public key %q for signature by %q", assert.SignKeyID(), assert.AuthorityID())
+		}
 		if err != nil {
-			return err
+			return fmt.Errorf("error finding matching public key for signature: %v", err)
+		}
+
+		now := time.Now()
```

Comment:
Could move this out of the if block and use the same value in the else. Removes a tiny bit of duplication.